float64 avg_abs_error
float64 max_abs_error
